### ISSUE TITLE
Investigate exception in logs when running the simulator as a web job

### ISSUE THREAD

**Issue by bennage** [mspnp/iot-journey#181]
I'm running the simulator (a version before the recent security changes in #169) as a long running web job. For the most part, it is behaving as expected. However, I see periodic errors that are related to the configuration.
I'd like to  
- [ ] see if the errors reproducible in the current master
- [ ] isolate the cause and correct it

The relevant log output is:
```
[06/30/2015 23:48:07 > 3ccdb3: SYS INFO] Status changed to PendingRestart
[06/30/2015 23:49:09 > 3ccdb3: SYS INFO] Run script 'WebJob.cmd' with script host - 'WindowsScriptHost'
[06/30/2015 23:49:09 > 3ccdb3: SYS INFO] Status changed to Running
[06/30/2015 23:49:09 > 3ccdb3: INFO] 
[06/30/2015 23:49:09 > 3ccdb3: INFO] D:\local\Temp\jobs\continuous\ScenarioSimulator\3cxj00lk.dvd>Microsoft.Practices.IoTJourney.ScenarioSimulator.ConsoleHost.exe /default /webjob 
[06/30/2015 23:49:10 > 3ccdb3: ERR ] 
[06/30/2015 23:49:10 > 3ccdb3: ERR ] Unhandled Exception: System.IO.FileNotFoundException: Could not load file or assembly 'Microsoft.WindowsAzure.ServiceRuntime, Version=2.5.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' or one of its dependencies. The system cannot find the file specified.
[06/30/2015 23:49:10 > 3ccdb3: ERR ]    at Microsoft.Practices.IoTJourney.ConfigurationHelper.TryGetAzureConfigValue[T](String key, T& val)
[06/30/2015 23:49:10 > 3ccdb3: ERR ]    at Microsoft.Practices.IoTJourney.ConfigurationHelper.GetConfigValue[T](String key) in c:\dev\iot-journey\src\Implementation\Shared\Core\ConfigurationHelper.cs:line 106
[06/30/2015 23:49:10 > 3ccdb3: ERR ]    at Microsoft.Practices.IoTJourney.ScenarioSimulator.SimulatorConfiguration.GetCurrentConfiguration() in c:\dev\iot-journey\src\Implementation\Simulator\ScenarioSimulator\SimulatorConfiguration.cs:line 27
[06/30/2015 23:49:10 > 3ccdb3: ERR ]    at Microsoft.Practices.IoTJourney.ScenarioSimulator.ConsoleHost.Program.<MainAsync>d__d.MoveNext() in c:\dev\iot-journey\src\RunFromConsole\ScenarioSimulator.ConsoleHost\Program.cs:line 28
[06/30/2015 23:49:10 > 3ccdb3: ERR ] --- End of stack trace from previous location where exception was thrown ---
[06/30/2015 23:49:10 > 3ccdb3: ERR ]    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
[06/30/2015 23:49:10 > 3ccdb3: ERR ]    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
[06/30/2015 23:49:10 > 3ccdb3: ERR ]    at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
[06/30/2015 23:49:10 > 3ccdb3: ERR ]    at Microsoft.Practices.IoTJourney.AsyncPump.Run(Func`1 func) in c:\dev\iot-journey\src\Implementation\Shared\Core\AsyncPump.cs:line 43
[06/30/2015 23:49:10 > 3ccdb3: ERR ]    at Microsoft.Practices.IoTJourney.ScenarioSimulator.ConsoleHost.Program.Main(String[] args) in c:\dev\iot-journey\src\RunFromConsole\ScenarioSimulator.ConsoleHost\Program.cs:line 21
[06/30/2015 23:49:10 > 3ccdb3: SYS ERR ] Job failed due to exit code -532462766
[06/30/2015 23:49:10 > 3ccdb3: SYS INFO] Process went down, waiting for 60 seconds
[06/30/2015 23:49:10 > 3ccdb3: SYS INFO] Status changed to PendingRestart
```

**Comment by lbrader** [comment]
Have the latest code and running the Simulator web job. I'll take a look at the 2 issues.

**Comment by lbrader** [comment]
@bennage - this issue occurs for every deployment from VS.  Digging down the actual error message is :

[FileNotFoundException: Could not load file or assembly 'Microsoft.WindowsAzure.ServiceRuntime, Version=2.5.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' or one of its dependencies. The system cannot find the file specified.]
   Microsoft.Practices.IoTJourney.ConfigurationHelper.TryGetAzureConfigValue(String key, T& val) +0
   Microsoft.Practices.IoTJourney.ConfigurationHelper.GetConfigValue(String key) +58
   DeviceProvisioning.AccessTokens.TokenProvider..ctor() +32
   DeviceProvisioning.AutoFacConfig.Configure() +106
   ProvisioningWebApi.WebApiApplication.Application_Start() +9
[HttpException (0x80004005): Could not load file or assembly 'Microsoft.WindowsAzure.ServiceRuntime, Version=2.5.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' or one of its dependencies. The system cannot find the file specified.]
   System.Web.HttpApplicationFactory.EnsureAppStartCalledForIntegratedMode(HttpContext context, HttpApplication app) +9916977
   System.Web.HttpApplication.RegisterEventSubscriptionsWithIIS(IntPtr appContext, HttpContext context, MethodInfo[] handlers) +118
   System.Web.HttpApplication.InitSpecial(HttpApplicationState state, MethodInfo[] handlers, IntPtr appContext, HttpContext context) +172
   System.Web.HttpApplicationFactory.GetSpecialApplicationInstance(IntPtr appContext, HttpContext context) +336
   System.Web.Hosting.PipelineRuntime.InitializeApplication(IntPtr appContext) +296

[HttpException (0x80004005): Could not load file or assembly 'Microsoft.WindowsAzure.ServiceRuntime, Version=2.5.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' or one of its dependencies. The system cannot find the file specified.]
   System.Web.HttpRuntime.FirstRequestInit(HttpContext context) +9930872
   System.Web.HttpRuntime.EnsureFirstRequestInit(HttpContext context) +101
   System.Web.HttpRuntime.ProcessRequestNotificationPrivate(IIS7WorkerRequest wr, HttpContext context) +254

**Comment by lbrader** [comment]
In researching out other implementation of Web Jobs.   Need some better links.  Here's the one I seen reference  the most. 
https://azure.microsoft.com/en-us/documentation/articles/websites-webjobs-resources/

From the reading what we have should work as a Web Job.  There are statements made that you can use a console application and not use the Web Job SDK.   And at one point I understood this worked.   So something has changed.

The code examples from the above link and Scott Hanselman example.
http://www.hanselman.com/blog/IntroducingWindowsAzureWebJobs.aspx

Both have the console main with the following Web Job SDK
        static void Main(string[] args)
        {
            JobHost host = new JobHost();
            host.RunAndBlock();
        }
 
I'll time box and sync up with Matt.

**Comment by MikeWasson** [comment]
Are we hitting http://stackoverflow.com/a/14203792/1261807?

**Comment by lbrader** [comment]
Good catch Mike.  Just sync up with @bennage and he found the issue with ConfigurationHelper.cs, which is a bit of legacy.  It was designed to support with the simulator as a cloud service.  

the solution is to remove all references to 'Microsoft.WindowsAzure.ServiceRuntime' from our Configuration code.   We are no longer supporting the Simulator as cloud service.  Instead we are supporting the Simulator as a Web Job to handle our scaling requirements.

**Comment by lbrader** [comment]
This will be fixed in #251
